### ISSUE TITLE
1.18 doesn't start on macOS/SunOS

### ISSUE THREAD

**Issue by domodwyer** [NLnetLabs/unbound#928]
**Describe the bug**
I was running Unbound 1.17.1 on macOS 13.5 with no issue, but a `brew update` shipped 1.18 to me today, and Unbound no longer starts.

It dies with:

```
fatal error: recvmsg: No support for IPV6_PKTINFO; IP_PKTINFO or IP_RECVDSTADDR. Please disable interface-automatic
```

I didn't have `interface-automatic` enabled (defaults to "no") so I added it to my config anyway, but no change.

**To reproduce**

Minimal config:

```
server:
	interface: 127.0.0.1@5000
	username: ""
	use-syslog: no
	interface-automatic: no
```

Run: `sudo unbound -c /tmp/min.conf -vvv -d`

And then: `dig google.com @127.0.0.1 -p 5000` which fails - when unbound receives the query it dies with the fatal error.

Server logs:

```
[1693502752] unbound[9085:0] notice: Start of unbound 1.18.0.
[1693502752] unbound[9085:0] debug: increased limit(open files) from 256 to 4140
[1693502752] unbound[9085:0] debug: creating udp4 socket 127.0.0.1 5000
[1693502752] unbound[9085:0] debug: creating tcp4 socket 127.0.0.1 5000
[1693502752] unbound[9085:0] warning: did not exit gracefully last time (8648)
[1693502752] unbound[9085:0] debug: chdir to /opt/homebrew/etc/unbound
[1693502752] unbound[9085:0] debug: chroot to /opt/homebrew/etc/unbound
[1693502752] unbound[9085:0] debug: switching log to stderr
[1693502752] unbound[9085:0] debug: module config: "validator iterator"
[1693502752] unbound[9085:0] notice: init module 0: validator
[1693502752] unbound[9085:0] debug: validator nsec3cfg keysz 1024 mxiter 150
[1693502752] unbound[9085:0] debug: validator nsec3cfg keysz 2048 mxiter 150
[1693502752] unbound[9085:0] debug: validator nsec3cfg keysz 4096 mxiter 150
[1693502752] unbound[9085:0] notice: init module 1: iterator
[1693502752] unbound[9085:0] debug: target fetch policy for level 0 is 3
[1693502752] unbound[9085:0] debug: target fetch policy for level 1 is 2
[1693502752] unbound[9085:0] debug: target fetch policy for level 2 is 1
[1693502752] unbound[9085:0] debug: target fetch policy for level 3 is 0
[1693502752] unbound[9085:0] debug: target fetch policy for level 4 is 0
[1693502752] unbound[9085:0] debug: donotq: 127.0.0.0/8
[1693502752] unbound[9085:0] debug: donotq: ::1
[1693502752] unbound[9085:0] debug: total of 59446 outgoing ports available
[1693502752] unbound[9085:0] debug: start threads
[1693502752] unbound[9085:0] debug: libevent 2.1.12-stable uses kqueue method.
[1693502752] unbound[9085:0] debug: no config, using builtin root hints.
[1693502752] unbound[9085:0] debug: cache memory msg=295704 rrset=295704 infra=30080 val=296032
[1693502752] unbound[9085:0] info: start of service (unbound 1.18.0).
[1693502754] unbound[9085:0] fatal error: recvmsg: No support for IPV6_PKTINFO; IP_PKTINFO or IP_RECVDSTADDR. Please disable interface-automatic
```

(note `interface-automatic` is already explicitly disabled)

**Expected behavior**
It should run.

**System:**
 - Unbound version: 1.18
 - OS: macOS 13.5
 - `unbound -V` output:

```
Version 1.18.0

Configure line: --prefix=/opt/homebrew/Cellar/unbound/1.18.0 --sysconfdir=/opt/homebrew/etc --enable-event-api --enable-tfo-client --enable-tfo-server --with-libevent=/opt/homebrew/opt/libevent --with-libnghttp2=/opt/homebrew/opt/libnghttp2 --with-ssl=/opt/homebrew/opt/openssl@3 --with-libexpat=/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr
Linked libs: libevent 2.1.12-stable (it uses kqueue), OpenSSL 3.1.2 1 Aug 2023
Linked modules: dns64 respip validator iterator
TCP Fastopen feature available

BSD licensed, see LICENSE in source package for details.
Report bugs to unbound-bugs@nlnetlabs.nl or https://github.com/NLnetLabs/unbound/issues
```

**Additional information**
Obtained via homebrew - even if it's a packaging issue, unbound should probably not recommend disabling something when it's already disabled.

**Comment by debohman** [comment]
I am seeing a similar result on 10.12.6:
```
Version 1.18.0

Configure line: --with-libnghttp2 --with-libevent
Linked libs: libevent 2.1.12-stable (it uses kqueue), OpenSSL 3.1.2 1 Aug 2023
Linked modules: dns64 respip validator iterator

BSD licensed, see LICENSE in source package for details.
Report bugs to unbound-bugs@nlnetlabs.nl or https://github.com/NLnetLabs/unbound/issues
```

**Comment by debohman** [comment]
Here are the logs I am seeing:
```
Aug 31 21:51:50 unbound[45673:0] warning: did not exit gracefully last time (45620)
Aug 31 21:51:50 unbound[45673:0] notice: init module 0: validator
Aug 31 21:51:50 unbound[45673:0] notice: init module 1: iterator
Aug 31 21:51:50 unbound[45673:0] info: start of service (unbound 1.18.0).
Aug 31 21:51:58 unbound[45673:0] fatal error: recvmsg: No support for IPV6_PKTINFO; IP_PKTINFO or IP_RECVDSTADDR. Please disable interface-automatic
Aug 31 21:51:58 unbound[45673:1] fatal error: recvmsg: No support for IPV6_PKTINFO; IP_PKTINFO or IP_RECVDSTADDR. Please disable interface-automatic
Aug 31 21:51:58 unbound[45673:3] fatal error: recvmsg: No support for IPV6_PKTINFO; IP_PKTINFO or IP_RECVDSTADDR. Please disable interface-automatic
```

**Comment by yvoinov** [comment]
Have similar issue (probably) on SunOS 5.10. However, without any log info and crashes. Unbound runs but stop responding short time after startup.

**Comment by rodrigobahiense** [comment]
I'm having exactly the same issue on macOS 13.5.1 (arm64) and unbound 1.18.0 installed via homebrew.

Reverting to 1.17.1 was the only way to get it working, as even using the default config for 1.18 made it generate the same error.

**Comment by pleblancq** [comment]
I confirm that I have the same issue as @domodwyer after an update to 1.18.0.

macOS: 11.7.9 (Big Sur on intel)

I reverted to 1.17.1 for now.

**Comment by marmolak** [comment]
Same here after unbound  update (new version: 1.18.0) on macOS 12.6.8.

```
unbound[26744:0] fatal error: recvmsg: No support for IPV6_PKTINFO; IP_PKTINFO or IP_RECVDSTADDR. Please disable interface-automatic
```

**Comment by alebcay** [comment]
I am seeing the same issue with 1.18.0 on macOS 12.6.8, _built from source tarball_ (not installed via Homebrew).

Performed `git bisect` between 1.18.0 and 1.17.1, first bad commit appears to be 2e6ddd603279cab26e3ee344a1ba26a7ba428e2c.

(Homebrew maintainer here, just chiming in since we received https://github.com/Homebrew/homebrew-core/issues/141643 downstream.)

**Comment by Philip-NLnetLabs** [comment]
Should be fixed in master now.

**Comment by domodwyer** [comment]
I can confirm HEAD is working a treat for me - thanks @Philip-NLnetLabs!

**Comment by yvoinov** [comment]
Works. Thanks, Phil.

**Comment by debohman** [comment]
The  fix seems to work on top of release 1.18.0.

Thank you.
